### ISSUE TITLE
How to Start?

### ISSUE THREAD

**Issue by BugDiver** [JunoLab/atom-ink#57]
I,m writing a atom package which supports [Gauge](getgauge.io). I want to use atom-ink package for it but it's [documentation](https://atom.io/packages/ink) is not good enough. It doesn't tell how to use this in our package and which kind of APIs are available. It would be great if  it's  provide a good docs.

**Comment by MikeInnes** [comment]
Yup this is definitely true. It's worth having a browse through the code because @pfitzseb has added some nice internal docs for at least the inline results portion.

You're also very welcome to ask any specific questions you have. What kind of features are you looking for? We're more than happy to point you in the right direction.

**Comment by carocad** [comment]
hey @MikeInnes, I understand that it can get difficult to document something so new and with so many functionalities.

I am currently working on replacing proto-repl's console with atom-ink's one. I managed to get a very basic console-editor working [(see gis)t](https://gist.github.com/carocad/6334a5d457b868e75da55f1b0233151c) but I would like to know if you could provide me some more info.
I am specially interested in the error display methods shown in this [gif](https://raw.githubusercontent.com/JunoLab/atom-ink/master/demos/full.gif). How do I get to highlight and link file-addresses + line-range to the editors? 
I tried `ink.highlight(atom.workspace.getActiveTextEditor(), 66, 69)` but it crashes :/

Any piece of information that you can provide would be helpful.

BTW: do you accept pull-request on documentation or is it still too early, too unstable to do that?

**Comment by pfitzseb** [comment]
Have a look at the implementation over at [atom-julia-client](https://github.com/JunoLab/atom-julia-client/blob/6c0d31456eb5019d8f5d3d472dee96a92bba0c93/lib/runtime/evaluation.coffee#L68-L77).

Documentation would be great, imo, especially by someone not as involved in the dev process -- so if you're feeling up to it... :) Also, feel free to ask any questions that arise in the meantime.

**Comment by carocad** [comment]
@pfitzseb thanks for the quick reply.

Sadly I had already looked at that method but it alone doesn't do the job :(
I tried

``` babel.js
ink.highlights.errorLines([
        {file: './atom-foo/atom-foo.js', line: 64},
        {file: 'home/camilo/Proyectos/atom-foo/atom-foo.js', line: 65}
      ])
```

but nothing happens.

I can identify the methods because I think they are very well named but there is no clear big picture of ink itself nor how to use it. I already have tried both methods that I think that could do the job but it seems that they alone are not enough so obviously I'm missing a detail that's hidden in the implementation itself.

**Comment by MikeInnes** [comment]
Most likely this isn't working because you haven't used the full path to the file, which should match `editor.getPath()`. Try calling that method to get the paths, or alternatively clicking the path in the status bar.

In general, if the API doesn't work in an intuitive way then you can consider that a bug on our side :)
